### PR TITLE
Expunge COFFEE

### PR DESCRIPTION
**Pull request by connorjward**
~Depends on https://github.com/inducer/loopy/issues/762~ resolved

### CODE REVIEW

File: tsfc/driver.py [hunk 1]
```diff
@@ -96,13 +107,8 @@ def compile_integral(integral_data, form_data, prefix, parameters, interface, co
...
     parameters = preprocess_parameters(parameters)
     if interface is None:
-        if coffee:
-            import tsfc.kernel_interface.firedrake as firedrake_interface_coffee
-            interface = firedrake_interface_coffee.KernelBuilder
-        else:
-            # Delayed import, loopy is a runtime dependency
-            import tsfc.kernel_interface.firedrake_loopy as firedrake_interface_loopy
-            interface = firedrake_interface_loopy.KernelBuilder
+        import tsfc.kernel_interface.firedrake_loopy as firedrake_interface_loopy
```
Comment (JDBetteridge) [inline tsfc/driver.py:110]: Move to top

File: tsfc/kernel_interface/firedrake_loopy.py [hunk 2]
```diff
@@ -99,6 +99,7 @@ def _coefficient(self, coefficient, name):
         """
```
Comment (JDBetteridge) [inline tsfc/kernel_interface/firedrake_loopy.py:99]: Add return type to docstring

File: tsfc/loopy.py [hunk 3]
```diff
@@ -199,7 +199,7 @@ def active_indices(mapping, ctx):
 
 
 def generate(impero_c, args, scalar_type, kernel_name="loopy_kernel", index_names=[],
-             return_increments=True, log=False):
+             return_increments=True, log=False, stop_early=False):
```
Comment (JDBetteridge) [inline tsfc/loopy.py:202]: Remove `stop_early`

**Review by JDBetteridge — CHANGES_REQUESTED** [review]

File: tests/test_tensor.py [hunk 4]
```diff
@@ -97,7 +95,9 @@ def form(cell, degree):
     flops = [count_flops(form(cell, int(degree)))
              for degree in degrees]
     rates = numpy.diff(numpy.log(flops)) / numpy.diff(numpy.log(degrees + 1))
-    assert (rates < order).all()
+    # FIXME This is failing without this change because the value is 3.16
```
Comment (connorjward) [inline tests/test_tensor.py:98]: Start at a higher degree instead.

**Review by JDBetteridge — APPROVED** [review]

**Pull request merged**